server/keyshare/keyshareserver/server.go

```diff
@@ -240,6 +298,71 @@ func (s *Server) generateCommitments(user *User, authorization string, keys []ir
 	return &irma.ProofPCommitmentMap{Commitments: mappedCommitments}, nil
 }
 
+// /api/v2/prove/getCommitments
+func (s *Server) handleCommitmentsV2(w http.ResponseWriter, r *http.Request) {
+	// Fetch from context
+	user := r.Context().Value("user").(*User)
+	authorization := r.Context().Value("authorization").(string)
+
+	// Read keys
+	var req irma.GetCommitmentsRequest
+	if err := server.ParseBody(r, &req); err != nil {
+		server.WriteError(w, server.ErrorInvalidRequest, err.Error())
+		return
+	}
+	if len(req.Keys) == 0 {
+		s.conf.Logger.Info("Malformed request: no keys for commitment specified")
+		server.WriteError(w, server.ErrorInvalidRequest, "No key specified")
+		return
+	}
+
+	commitments, err := s.generateCommitmentsV2(user, authorization, req)
+	// TODO: can ErrInvalidChallenge be removed?
+	if err != nil && (err == keysharecore.ErrInvalidChallenge || err == keysharecore.ErrInvalidJWT) {
+		server.WriteError(w, server.ErrorInvalidRequest, err.Error())
+		return
+	}
```

Comment:
Is this something you want to sort out in a later PR?

Reply:
yep